components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/execution/PreUpdatePasswordActionRequestBuilder.java

```diff
@@ -0,0 +1,219 @@
+import org.wso2.carbon.utils.Secret;
+import org.wso2.carbon.utils.UnsupportedSecretTypeException;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+
+/**
+ * This class is responsible for building the action execution request for the pre update password action.
+ */
+public class PreUpdatePasswordActionRequestBuilder implements ActionExecutionRequestBuilder {
+
+    private UserActionContext userActionContext;
+    private PreUpdatePasswordAction preUpdatePasswordAction;
+
+    @Override
+    public ActionType getSupportedActionType() {
+
+        return ActionType.PRE_UPDATE_PASSWORD;
+    }
+
+    @Override
+    public ActionExecutionRequest buildActionExecutionRequest(Map<String, Object> eventContext)
+            throws ActionExecutionRequestBuilderException {
+
+        resolveAction(eventContext);
+        resolveUserActionContext(eventContext);
+
+        ActionExecutionRequest.Builder actionRequestBuilder = new ActionExecutionRequest.Builder();
+        actionRequestBuilder.actionType(getSupportedActionType());
+        actionRequestBuilder.event(getEvent());
+
+        return actionRequestBuilder.build();
+    }
+
+    private void resolveUserActionContext(Map<String, Object> eventContext)
+            throws ActionExecutionRequestBuilderException {
+
+        Object action = eventContext.get(PreUpdatePasswordActionConstants.USER_ACTION_CONTEXT);
+        if (!(action instanceof UserActionContext)) {
+            throw new ActionExecutionRequestBuilderException("User Action Context cannot be null.");
+        }
+        userActionContext = (UserActionContext) action;
+    }
+
+    private void resolveAction(Map<String, Object> eventContext) throws ActionExecutionRequestBuilderException {
+
+        Object action = eventContext.get("action");
+        if (!(action instanceof PreUpdatePasswordAction)) {
+            throw new ActionExecutionRequestBuilderException("Pre Update Password Action cannot be null.");
+
+        }
+        preUpdatePasswordAction = (PreUpdatePasswordAction) action;
+    }
+
+    private Event getEvent() throws ActionExecutionRequestBuilderException {
+
+        return new PreUpdatePasswordEvent.Builder()
+                .initiator(getInitiator())
+                .action(getAction())
+                .tenant(getTenant())
+                .user(getUser())
+                .userStore(new UserStore(userActionContext.getUserStoreDomain()))
+                .build();
+    }
+
+    private PreUpdatePasswordEvent.FlowInitiator getInitiator() throws ActionExecutionRequestBuilderException {
+
+        Flow flow = IdentityContext.getThreadLocalIdentityContext().getFlow();
+        if (flow == null) {
+            throw new ActionExecutionRequestBuilderException("Flow is null in the Identity Context.");
```

Comment:
Let's not say flow is null. Let's say flow is not identified/populated etc

Reply:
Fixed